crm_compassion/model/event_compassion.py

```diff
@@ -0,0 +1,307 @@
+            'view_type': 'form',
+            'res_model': 'project.task',
+            'src_model': 'crm.event.compassion',
+            'context': {
+                'search_default_project_id': [project_id],
+                'default_project_id': project_id,
+                'active_test': False},
+            'search_view_id': self.env['ir.model.data'].get_object_reference(
+                'project', 'view_task_search_form')[1]
+        }
+
+    ##########################################################################
+    #                             PRIVATE METHODS                            #
+    ##########################################################################
+    def _get_project_vals(self):
+        """ Creates a new project based on the event.
+        """
+        members = self.env['res.users'].search(
+            [('partner_id', 'in', self.staff_ids.ids)]).ids
+        return {
+            'name': self.full_name,
+            'use_tasks': True,
+            'analytic_account_id': self.analytic_id.id,
+            'project_type': self.type,
+            'user_id': self.user_id.id,
+            'partner_id': self.partner_id.id,
+            'members': [(6, 0, members)],   # many2many field
+            'date_start': self.start_date,
+            'date': self.end_date,
+            'state': 'open',
+        }
+
+    def _get_analytic_vals(self):
+        name = self.name
+        parent_id = self.parent_id and self.parent_id.id
+        if self.city:
+            name += ' ' + self.city
+        if not parent_id:
+            parent_id = self._find_parent_analytic()
+            super(event_compassion, self).write({'parent_id': parent_id})
+        return {
+            'name': name,
+            'type': 'event',
+            'event_type': self.type,
+            'date_start': self.start_date,
+            'date': self.end_date,
+            'parent_id': parent_id,
+            'use_timesheets': True,
+            'partner_id': self.partner_id.id,
+            'manager_id': self.user_id.id,
+            'user_id': self.user_id.id,
+        }
+
+    def _get_origin_vals(self, analytic_id):
+        return {
+            'type': 'event',
+            'event_id': self.id,
+            'analytic_id': analytic_id,
+        }
+
+    def _find_parent_analytic(self):
+        analytics_obj = self.env['account.analytic.account']
+        categ_id = analytics_obj.search(
+            [('name', 'ilike', self.type)])[0].id
+        acc_ids = analytics_obj.search(
+            [('name', '=', self.year), ('parent_id', '=', categ_id)]).ids
+        if not acc_ids:
+            # The category for this year does not yet exist
+            acc_ids = [analytics_obj.create({
+                'name': self.year,
+                'type': 'view',
+                'code': 'AA' + self.type[:2].upper() + self.year,
+                'parent_id': categ_id
+            }).id]
+        return acc_ids[0]

```

Comment:
`acc_ids = [analytics_obj.create({...}).id]` is not really clear. Maybe rename `acc_ids` in `acc_id`, use parameter `limit=1` in the search and remove the brackets `acc_id = analytics_obj.create({...}).id`